Library/Homebrew/cask/cask_loader.rb

```diff
@@ -113,8 +114,22 @@ def self.try_new(ref, warn: false)
         return unless path.expand_path.exist?
         return if invalid_path?(path)
 
-        return if Homebrew::EnvConfig.forbid_packages_from_paths? &&
-                  !path.realpath.to_s.start_with?("#{Caskroom.path}/", "#{HOMEBREW_LIBRARY}/Taps/")
+        allowed_paths = ["#{Caskroom.path}/", "#{HOMEBREW_LIBRARY}/Taps/"]
+        validation_result = Utils::Path.validate_package_path(
+          path,
+          allowed_paths,
+          "casks",
+          "brew create --cask <url> --tap=<user|org>/<repository>"
```

Comment:
We could probably infer whether to add `--cask` based on whether the type is `casks` or `formulae`